v4/weaviate/weaviate_client.go

```diff
@@ -34,6 +35,19 @@ type Config struct {
 	Headers map[string]string
 }
 
+func NewConfig(Host string, Scheme string, AuthConfig auth.Config, Headers map[string]string) (*Config, error) {
+	var client *http.Client = nil
+	var err error
+	if AuthConfig != nil {
```

Comment:
I would make `AutoConfig` and `Headers` variable names to start with lower letter